packages/flutter/lib/foundation.dart

```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// @dart = 2.8
+// @dart = 2.9
```

Comment:
I think probably how this should be landed is by bumping the min sdk version in the pubspec to be a 2.9 release, and removing the // @dart comments in the opted in files entirely.  @jakemac53 @jonahwilliams does that sound right?

Reply:
That sounds like an approach with less churn. Is 2.9 _the_ null safety language version, or could that change?

Reply:
2.9 will *not* be the final null safety language version.  It is likely, but not guaranteed that 2.10 will be the final null safety language version.

Reply:
okay, in that case we should definitely do it in the pubspec so there is only a single location, right?

Reply:
I prefered to explicitly use `// @dart = 2.9` during the incremental migration for simple searchability. Once all framework is migrated I will remove it. Note that the [pubspec.yaml sdk version is already pointing to 2.9](https://github.com/flutter/flutter/blob/master/packages/flutter/pubspec.yaml#L8).

Does it look good to you?

Reply:
If it makes the process easier, I would say to use it - but file an issue to clean up the overrides once it has all landed.

Reply:
I think that's better, and it's how pub packages need to migrate (because otherwise you're allowing sdks that don't support null safety, which won't work).  I think with flutter, because you provide your own copy of the sdk, you don't run into the versioning problem, but it still seems like the right approach to me.  But I don't know all the details of how package:flutter works - it's Not Like Other Packages.

Reply:
Ok, works for me.  Note that if/when the Dart version bumps to a 2.10 pre-release, there will a bit of a bump - you won't be able roll 2.10 in until you change all of these from // @dart = 2.9 to // @dart = 2.10.

Reply:
could we set it to 2.10 now?

Reply:
> could we set it to 2.10 now?

You could try it, but I wouldn't really recommend it.  You're more or less relying on undefined behavior in that case.  It might just work, in which case it's probably fine, but it also might work in some places and not others, or suddenly stop working, or whatever.

Reply:
If this is really just about tracking which things are migrated, maybe consider just using some other arbitrary string to tag them?  // @dart = migrated ?

Reply:
Ok I'll remove it directly. I don't use it too much now.